testing.rst

```diff
@@ -876,164 +983,16 @@ their type::
-            disable_delivery: true
-
-    .. code-block:: xml
-
-        <!-- config/packages/test/swiftmailer.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xmlns:swiftmailer="http://symfony.com/schema/dic/swiftmailer"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                https://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/swiftmailer
-                https://symfony.com/schema/dic/swiftmailer/swiftmailer-1.0.xsd">
-
-            <!-- ... -->
-            <swiftmailer:config disable-delivery="true"/>
-        </container>
-
-    .. code-block:: php
-
-        // config/packages/test/swiftmailer.php
-
-        // ...
-        $container->loadFromExtension('swiftmailer', [
-            'disable_delivery' => true,
-        ]);
-
-You can also use a different environment entirely, or override the default
-debug mode (``true``) by passing each as options to the ``createClient()``
-method::
-
-    $client = static::createClient([
-        'environment' => 'my_test_env',
-        'debug'       => false,
-    ]);
-
-Customizing Database URL / Environment Variables
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-If you need to customize some environment variables for your tests (e.g. the
-``DATABASE_URL`` used by Doctrine), you can do that by overriding anything you
-need in your ``.env.test`` file:
-
-.. code-block:: text
-
-    # .env.test
-    DATABASE_URL="mysql://db_user:db_password@127.0.0.1:3306/db_name_test?serverVersion=5.7"
-
-    # use SQLITE
-    # DATABASE_URL="sqlite:///%kernel.project_dir%/var/app.db"
-
-This file is automatically read in the ``test`` environment: any keys here override
-the defaults in ``.env``.
-
-.. caution::
-
-    Applications created before November 2018 had a slightly different system,
-    involving a ``.env.dist`` file. For information about upgrading, see:
-    :doc:`configuration/dot-env-changes`.
-
-Sending Custom Headers
-~~~~~~~~~~~~~~~~~~~~~~
-
-If your application behaves according to some HTTP headers, pass them as the
-second argument of ``createClient()``::
-
-    $client = static::createClient([], [
-        'HTTP_HOST'       => 'en.example.com',
-        'HTTP_USER_AGENT' => 'MySuperBrowser/1.0',
-    ]);
-
-You can also override HTTP headers on a per request basis::
-
-    $client->request('GET', '/', [], [], [
-        'HTTP_HOST'       => 'en.example.com',
-        'HTTP_USER_AGENT' => 'MySuperBrowser/1.0',
-    ]);
-
-.. tip::
-
-    The test client is available as a service in the container in the ``test``
-    environment (or wherever the :ref:`framework.test <reference-framework-test>`
-    option is enabled). This means you can override the service entirely
-    if you need to.
-
-.. index::
-   pair: PHPUnit; Configuration
+TODO
```

Comment:
Could we also add a section about [`ApiTestCase`](https://api-platform.com/docs/distribution/testing/#testing-the-api) (or a link to this page)?

Reply:
Yes, let's do that in a follow-up PR :)